src/main/java/attendance/domain/Attendances.java

```diff
@@ -1,38 +1,89 @@
 package attendance.domain;
 
 import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 import attendance.exception.AttendanceArgumentException;
 
-public record Attendances(Map<LocalDate, Attendance> attendances) {
-    public static final String DUPLICATE_DATE = "이미 출석되었습니다. 수정 기능을 이용해주세요.";
+public record Attendances(Map<LocalDate, Attendance> attendances, SystemDateTime systemDateTime) {
+    private static final String CANNOT_ATTENDANCE_WEEKEND_FORMAT = "MM월 dd일 E요일은 등교일이 아닙니다.";
+    private static final String CANNOT_FIND_ATTENDANCE = "출석 기록이 없습니다.";
+    private static final String DUPLICATE_ATTENDANCE = "이미 출석하였습니다. 수정 기능을 이용해주세요.";
 
-    public Attendances() {
-        this(new HashMap<>());
+    public Attendances(SystemDateTime systemDateTime) {
+        this(new HashMap<>(), systemDateTime);
     }
 
-    public void add(Attendance attendance) {
-        attendances.put(attendance.dateTime().toLocalDate(), attendance);
+    public void addAttendance(LocalDateTime dateTime) {
+        LocalDate date = dateTime.toLocalDate();
+        isDuplicateAttendance(date);
+        isValidateSchedule(date);
+        attendances.put(date, new Attendance(dateTime));
     }
 
-    public Optional<Attendance> findAttendance(LocalDate date) {
+    private void isDuplicateAttendance(LocalDate date) {
+        Optional<Attendance> attendance = getOptionalAttendance(date);
+        if (attendance.isPresent()) {
+            throw new AttendanceArgumentException(DUPLICATE_ATTENDANCE);
+        }
```

Comment:
```suggestion
        if (attendances.containsKey(date)) {
            throw new AttendanceArgumentException(DUPLICATE_ATTENDANCE);
        }
```
Optional이 불필요하게 사용되고 있네요.